node/network/statement-distribution/src/lib.rs

```diff
@@ -1545,15 +1554,30 @@ impl StatementDistribution {
 	/// Handle messages from responder background task.
 	async fn handle_responder_message(
 		&self,
+		peers: &HashMap<PeerId, PeerData>,
 		active_heads: &mut HashMap<Hash, ActiveHeadData>,
 		message: ResponderMessage,
 	) -> SubsystemResult<bool> {
 		match message {
 			ResponderMessage::GetData {
+				requesting_peer,
 				relay_parent,
 				candidate_hash,
 				tx,
 			} => {
+				if !requesting_peer_knows_about_candidate(
+					peers,
+					&requesting_peer,
+					&relay_parent,
+					&candidate_hash
+				) {
+					tracing::warn!(
```

Comment:
I'm not sure about `warn!` level, we seem to use `debug!` e.g. when a peer sends a duplicate message, which is also unexpected.

Reply:
I could not find a reason for this happening, except me writing buggy code or malicious activity .. and the latter would justify a `warn`... That was at least the reasoning for choosing `warn` here.